pkg/querier/queryrange/codec.go

```diff
@@ -744,23 +745,72 @@ func (Codec) MergeResponse(responses ...queryrangebase.Response) (queryrangebase
 		}, nil
 	case *VolumeResponse:
 		resp0 := responses[0].(*VolumeResponse)
-		headers := resp0.Headers
+		headers := make([]*definitions.PrometheusResponseHeader, len(resp0.Headers))
+		for i, header := range resp0.Headers {
```

Comment:
Why copy all the headers? Does something change them later on?

Reply:
Not sure I follow. This is doing the same thing as the old code, just turning the slice into a slice of pointers since that's what the `PrometheusReponse` wants. Is the suggestions here just to take `resp0.Headers[0]`?